src/main/java/org/opensearch/performanceanalyzer/rca/configs/SearchBackPressureRcaConfig.java

```diff
@@ -0,0 +1,137 @@
+                        DEFAULT_MAX_HEAP_INCREASE_THRESHOLD,
+                        (s) -> s >= 0 && s <= 100,
+                        Integer.class);
+        maxShardHeapCancellationPercentageThreshold =
+                conf.readRcaConfig(
+                        CONFIG_NAME,
+                        RCA_CONF_KEY_CONSTANTS.MAX_SHARD_HEAP_CANCELLATION_PERCENTAGE_FIELD,
+                        DEFAULT_SHARD_MAX_HEAP_CANCELLATION_THRESHOLD,
+                        (s) -> s >= 0 && s <= 100,
+                        Integer.class);
+        maxTaskHeapCancellationPercentageThreshold =
+                conf.readRcaConfig(
+                        CONFIG_NAME,
+                        RCA_CONF_KEY_CONSTANTS.MAX_TASK_HEAP_CANCELLATION_PERCENTAGE_FIELD,
+                        DEFAULT_TASK_MAX_HEAP_CANCELLATION_THRESHOLD,
+                        (s) -> s >= 0 && s <= 100,
+                        Integer.class);
+        minHeapDecreasePercentageThreshold =
+                conf.readRcaConfig(
+                        CONFIG_NAME,
+                        RCA_CONF_KEY_CONSTANTS.MAX_HEAP_USAGE_DECREASE_FIELD,
+                        DEFAULT_MIN_HEAP_DECREASE_THRESHOLD,
+                        (s) -> s >= 0 && s <= 100,
+                        Integer.class);
+        minShardHeapCancellationPercentageThreshold =
+                conf.readRcaConfig(
+                        CONFIG_NAME,
+                        RCA_CONF_KEY_CONSTANTS.MIN_SHARD_HEAP_CANCELLATION_PERCENTAGE_FIELD,
+                        DEFAULT_SHARD_MIN_HEAP_CANCELLATION_THRESHOLD,
+                        (s) -> s >= 0 && s <= 100,
+                        Integer.class);
+        minTaskHeapCancellationPercentageThreshold =
+                conf.readRcaConfig(
+                        CONFIG_NAME,
+                        RCA_CONF_KEY_CONSTANTS.MIN_TASK_HEAP_CANCELLATION_PERCENTAGE_FIELD,
+                        DEFAULT_TASK_MIN_HEAP_CANCELLATION_THRESHOLD,
+                        (s) -> s >= 0 && s <= 100,
+                        Integer.class);
+    }
+
+    // Getters for private field
+    public Integer getMaxHeapIncreasePercentageThreshold() {
+        return maxHeapIncreasePercentageThreshold;
+    }
+
+    public Integer getMaxShardHeapCancellationPercentageThreshold() {
+        return maxShardHeapCancellationPercentageThreshold;
+    }
+
+    public Integer getMaxTaskHeapCancellationPercentageThreshold() {
+        return maxTaskHeapCancellationPercentageThreshold;
+    }
+
+    public Integer getMinHeapDecreasePercentageThreshold() {
+        return minHeapDecreasePercentageThreshold;
+    }
+
+    public Integer getMinShardHeapCancellationPercentageThreshold() {
+        return minShardHeapCancellationPercentageThreshold;
+    }
+
+    public Integer getMinTaskHeapCancellationPercentageThreshold() {
+        return minTaskHeapCancellationPercentageThreshold;
+    }
+
+    // name for the configuration field
+    public static class RCA_CONF_KEY_CONSTANTS {
```

Comment:
It might be better to use `enum` instead of `public static class` here to define a list of related constants.
In general, enums offer better readability, type safety, and if we need to iterating over those constants we can simply use the `values()` method. This is more straightforward than manually iterating over constants in a public static final class.
you can do something like:
```
    public enum RCA_CONF_KEY_CONSTANTS {
        MAX_HEAP_USAGE_INCREASE_FIELD("max-heap-usage-increase"),
        MAX_SHARD_HEAP_CANCELLATION_PERCENTAGE_FIELD("max-shard-heap-cancellation-percentage");
        // ...

        private final String value;
        RCA_CONF_KEY_CONSTANTS(final String value) {
            this.value = value;
        }
        @Override
        public String toString() {
            return value;
        }
    }
```

Reply:
Thank you so much for the suggestion. I have switched the design from static class to enum